org.eclipse.lsp4e/src/org/eclipse/lsp4e/DocumentContentSynchronizer.java

```diff
@@ -225,10 +232,27 @@ private boolean serverSupportsWillSaveWaitUntil() {
 		return false;
 	}
 
-	private static final int WILL_SAVE_WAIT_UNTIL_TIMEOUT_IN_SECONDS = 5;
+	private static final String WILL_SAVE_WAIT_UNTIL_TIMEOUT__KEY = "timeout.willSaveWaitUntil"; //$NON-NLS-1$
+
 	private static final int WILL_SAVE_WAIT_UNTIL_COUNT_THRESHOLD = 3;
 	private static final Map<String, Integer> WILL_SAVE_WAIT_UNTIL_TIMEOUT_MAP = new ConcurrentHashMap<>();
 
+	/**
+	 * Converts a language server ID to the preference ID to define a timeout
+	 * for willSaveWaitUntil
+	 *
+	 * @return language server's preference ID to define a timeout for willSaveWaitUntil
+	 */
+	public static @NonNull String lsToWillSaveWaitUntilTimeoutKey(String serverId) {
```

Comment:
Instead of a serverId as parameter, the method should directly take the languageServerWrapper as argument here. Stronger typing helps.
Also, it shouldn't have to be public, should it?

Reply:
removed

Reply:
I'm still seeing it ;)

Reply:
now :)